apps/elections/components/AreaPicker.js

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import Select from 'react-select';
+
+import { getElectoralDistricts, getMunicipalities, getPollingDistricts } from './Backend.js';
+import { AreaType } from 'election';
+
+"use strict";
+
+// Individual selector statuses
+const NOT_SELECTED = "NOT_SELECTED";
+const LOADING = "LOADING";
+const DISABLED = "DISABLED";
+const SELECTED = "SELECTED";
+
+// Data constructors for all possible states of individual selector
+let selector = {
+  disabled: function() {
+    return {
+      status: DISABLED,
+      selection: null,
+      options: []
+    }
+  },
+  loading: function() {
+    return {
+      status: LOADING,
+      selection: null,
+      options: []
+    }
+  },
+  notSelected: function(options) {
+    return {
+      status: NOT_SELECTED,
+      selection: null,
+      options
+    }
+  },
+  selected: function(selection, options) {
+    return {
+      status: SELECTED,
+      selection,
+      options
+    }
+  }
+}
+
+let Selector = function(props) {
+  return (
+    <Select
+      value       = { props.state.selection }
+      options     = { props.state.options }
+      isDisabled  = { props.state.status == DISABLED }
+      isLoading   = { props.state.status == LOADING }
+      onChange    = { props.onChange }
+      isClearable = { true }
+      loadingMessage = { () => props.loadingMessage }
+      placeholder    = { props.placeholder }
+    />
+  )
+}
+
+export default class AreaPicker extends React.Component {
+  // called when creating the component
+  constructor(props) {
+    super(props);
+    // initially all selectors are disabled
+    this.state = {
+      [AreaType.ELECTORAL_DISTRICT]: selector.disabled(),
+      [AreaType.MUNICIPALITY]: selector.disabled(),
+      [AreaType.POLLING_DISTRICT]: selector.disabled(),
+    }
+  }
+  // called when component is loaded
+  componentDidMount() {
+    // load top-level areas for the first selector
+    this.getSelectorOptions(AreaType.ELECTORAL_DISTRICT);
+  }
+  render() {
+    return (
+      <div>
+        <Selector
+          state    = { this.state[AreaType.ELECTORAL_DISTRICT] }
+          onChange = { (option) => this.onSelection(AreaType.ELECTORAL_DISTRICT, option) }
+          placeholder    = "Select electoral district…"
+          loadingMessage = "Getting electoral districts…"
+        />
+        <Selector
+          state    = { this.state[AreaType.MUNICIPALITY] }
+          onChange = { (option) => this.onSelection(AreaType.MUNICIPALITY, option) }
+          placeholder    = "Select municipality…"
+          loadingMessage = "Getting municipalities…"
+        />
+        <Selector
+          state    = { this.state[AreaType.POLLING_DISTRICT] }
+          onChange = { (option) => this.onSelection(AreaType.POLLING_DISTRICT, option) }
+          placeholder    = "Select polling district…"
+          loadingMessage = "Getting polling districts…"
```

Comment:
The `placeholder`s and `loadingMessage`s  need to be Swedenalized

Reply:
I'll fix those